fix entitySlot entities under PrefabPlaceholders

killzoms opened the pull request:
fixes #1948

Jannify reviewed (commented):
LGTM CW
Needs in game testing with rejoining and server restarting

killzoms commented:
rebased on the latest changes and fixed a couple of minor issues. Floating Stones should look and function as they typically do in Vanilla Subnautica now. Fragments aren't 100% vanilla yet as there is currently no filtering based on finished Scan progress, that is out of scope for this PR.

Jannify commented on NitroxClient/GameLogic/Spawning/WorldEntities/PlaceholderGroupWorldEntitySpawner.cs:
```diff
@@ -121,6 +121,11 @@ public IEnumerator SpawnAsync(WorldEntity entity, Optional<GameObject> parent, E
             }
         }
 
+        if (prefabPlaceholderGroup.OnPrefabGroupSpawned != null) // Handles setting isKinematic on Floating Stones
+        {
+            prefabPlaceholderGroup.OnPrefabGroupSpawned();
+        }
```
Very minor
```suggestion
        prefabPlaceholderGroup.OnPrefabGroupSpawned?.Invoke(); // Handles setting isKinematic on Floating Stones
```

Jannify reviewed (approved):
LGTM CW

Coding-Hen reviewed (approved):
LG CW

dartasen reviewed (changes requested):
Need InGame testing
